ethcore/sync/src/chain/propagator.rs

```diff
@@ -96,13 +91,45 @@ impl SyncPropagator {
 		sent
 	}
 
+	/// propagate new known pool hashes to all peers
+	pub fn propagate_new_pooled_tx_hashes(sync: &mut ChainSync, io: &mut dyn SyncIo) -> usize {
+		/// Recommended limit from the spec.
+		const LIMIT: usize = 4096;
+
+		trace!(target: "sync", "Sending NewPooledTransactionsHashes to {:?}", sync.peers.keys());
+		let mut affected_peers = 0;
+		for (peer_id, peer) in &mut sync.peers {
+			let mut affected = false;
+			let mut packet = RlpStream::new();
+			packet.begin_unbounded_list();
+			if let Some(s) = &mut peer.unsent_pooled_hashes {
```

Comment:
As I said earlier we should be using `last_sent_hashes` here as well, we already have similar logic in transaction propagation.

Actually this whole method might not be needed at all. We should use current transaction propagation logic and this new ETH65 consolidation protocol should only be used to LIMIT the transactions we are sending to  hashes that were actually requested (or not seen) by the other peer.